tools/ctl/command/runbenchmark/runbenchmark.go

```diff
@@ -102,6 +108,70 @@ func parseArgs(cfg *Config, args []string) error {
 	return nil
 }
 
+// compileToFS builds a Go program and loads the binary directly into the provided memfs.
+func compileToFS(ctx context.Context, fs billy.Filesystem, srcPath string) error {
+	if filepath.Base(srcPath) == "main.go" {
+		return errors.New("unexpected main.go, did you mean to provide the dir?")
+	}
+	dir, err := os.MkdirTemp("", "*")
+	if err != nil {
+		return errors.Wrap(err, "creating tempdir")
+	}
+	defer os.RemoveAll(dir)
+	log.Printf("Compiling  %s...", srcPath)
+	cmd := exec.CommandContext(ctx, "go", "build", "-o", dir, "-gcflags", "-l=4", srcPath)
+	cmd.Env = append(os.Environ(), "GOOS=linux", fmt.Sprintf("GOARCH=%s", runtime.GOARCH), "CGO_ENABLED=0")
```

Comment:
doesn't appending to this have some side-effect? i think the annoying but side-effect-free version is []string{...}, os.Environ()...